nowcasting_dataloader/fake.py

```diff
@@ -38,6 +47,13 @@ def __getitem__(self, idx):
         Returns: Dictionary of random data
 
         """
-        x: BatchML = BatchML.fake(configuration=self.configuration)
-
-        return x.dict()
+        if self.add_position_encoding:
+            batch: Batch = Batch.fake(configuration=self.configuration)
```

Comment:
i wonder if there is a neater way to do this, 
is there an any way 'generate_position_encodings_for_batch' could work of BatchML?

Reply:
It would require quite a bit of rewriting, but it probably could, I think originally BatchML didn't have all the information that was needed, but I don't quite remember why right now